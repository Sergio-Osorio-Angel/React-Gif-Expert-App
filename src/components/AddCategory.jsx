import React, { useState } from 'react';
import PropTypes from "prop-types";

export function AddCategory ({onNewCategory}) {

    const [inputValue, setinputValue] = useState('');
    

    function onInputChange (event) {
        setinputValue(event.target.value)
    }

    function onSubmit (event)  {
        event.preventDefault();
        if (inputValue.trim().length <= 1) {
            return ;
        }
        onNewCategory(inputValue.trim());
        setinputValue('');
    }

    return (
        <form onSubmit={onSubmit} data-testid='form-intro'>
            {/* <input type="text" placeholder='Buscar Gifs' value={inputValue} onChange={(event)=> {onInputChange(event)}}/> */}
            <input type="text" placeholder='Buscar Gifs' value={inputValue} onChange={onInputChange}/>
        </form>
    )
}
AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
  }