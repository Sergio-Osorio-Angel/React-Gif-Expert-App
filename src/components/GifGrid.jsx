import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export function GifGrid({category}) {
 

  const {images, isLoading} = useFetchGifs(category);

  return (
    <>
        <h3>{category}</h3>
        {
          isLoading && (<span>Cargando...</span>)
        }
        <div className="card-grid">
          {
            // images.map((img) => (<GifGridItem key={img.id} title={img.title} url={img.url}></GifGridItem>))
            images.map((img) => (<GifGridItem key={img.id} {...img}></GifGridItem>))
          }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}