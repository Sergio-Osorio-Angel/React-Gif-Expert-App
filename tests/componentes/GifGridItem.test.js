import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('GifGridItem.jsx', () => {
    const title = 'Demon Slayer';
    const url = 'https://imagenes/test.jpg';

    test('match con Snapshot', () => {
        const {container} = render(<GifGridItem title={title} url={url}></GifGridItem>)
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifGridItem title={title} url={url}></GifGridItem>);
        const img = screen.getByRole('img');
        expect(img.src).toBe(url);
        expect(img.alt).toBe(title);
    })

    test('debe de mostar el titulo en el componente', () => {
        render(<GifGridItem title={title} url={url}></GifGridItem>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})