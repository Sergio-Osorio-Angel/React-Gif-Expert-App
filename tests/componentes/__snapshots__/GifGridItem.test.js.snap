// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem.jsx match con Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Demon Slayer"
      src="https://imagenes/test.jpg"
    />
    <p>
      Demon Slayer
    </p>
  </div>
</div>
`;
